From 1b5f608262e544e3c5c9382f69f9ba310f4af405 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Fri, 11 Jan 2019 01:01:52 +0100
Subject: [PATCH] Fix build against poppler-0.72.0

The relevant change in poppler 0.72 release notes:
* Rename GooString::getCString to GooString::c_str
---
 src/extension/internal/pdfinput/pdf-input.cpp |  4 ++++
 .../internal/pdfinput/pdf-parser.cpp          | 19 ++++++++++++++++---
 .../internal/pdfinput/svg-builder.cpp         |  8 ++++++++
 3 files changed, 28 insertions(+), 3 deletions(-)

diff --git a/src/extension/internal/pdfinput/pdf-input.cpp b/src/extension/internal/pdfinput/pdf-input.cpp
index 9621e30..ae29551 100644
--- a/src/extension/internal/pdfinput/pdf-input.cpp
+++ b/src/extension/internal/pdfinput/pdf-input.cpp
@@ -319,7 +319,11 @@ PdfImportDialog::PdfImportDialog(PDFDoc *doc, const gchar */*uri*/)
     _render_thumb = true;
 
     // Create PopplerDocument
+#if POPPLER_CHECK_VERSION(0,72,0)
+    Glib::ustring filename = _pdf_doc->getFileName()->c_str();
+#else
     Glib::ustring filename = _pdf_doc->getFileName()->getCString();
+#endif
     if (!Glib::path_is_absolute(filename)) {
         filename = Glib::build_filename(Glib::get_current_dir(),filename);
     }
diff --git a/src/extension/internal/pdfinput/pdf-parser.cpp b/src/extension/internal/pdfinput/pdf-parser.cpp
index 4a7fd8a..d4f746a 100644
--- a/src/extension/internal/pdfinput/pdf-parser.cpp
+++ b/src/extension/internal/pdfinput/pdf-parser.cpp
@@ -2381,9 +2381,14 @@ void PdfParser::opSetFont(Object args[], int /*numArgs*/)
   }
   if (printCommands) {
     printf("  font: tag=%s name='%s' %g\n",
-	   font->getTag()->getCString(),
-	   font->getName() ? font->getName()->getCString() : "???",
-	   args[1].getNum());
+#if POPPLER_CHECK_VERSION(0,72,0)
+          font->getTag()->c_str(),
+          font->getName() ? font->getName()->c_str() : "???",
+#else
+          font->getTag()->getCString(),
+          font->getName() ? font->getName()->getCString() : "???",
+#endif
+          args[1].getNum());
     fflush(stdout);
   }
 
@@ -2639,7 +2644,11 @@ void PdfParser::doShowText(GooString *s) {
     double lineX = state->getLineX();
     double lineY = state->getLineY();
     oldParser = parser;
+#if POPPLER_CHECK_VERSION(0,72,0)
+    p = s->c_str();
+#else
     p = s->getCString();
+#endif
     len = s->getLength();
     while (len > 0) {
       n = font->getNextChar(p, len, &code,
@@ -2694,7 +2703,11 @@ void PdfParser::doShowText(GooString *s) {
 
   } else {
     state->textTransformDelta(0, state->getRise(), &riseX, &riseY);
+#if POPPLER_CHECK_VERSION(0,72,0)
+    p = s->c_str();
+#else
     p = s->getCString();
+#endif
     len = s->getLength();
     while (len > 0) {
       n = font->getNextChar(p, len, &code,
diff --git a/src/extension/internal/pdfinput/svg-builder.cpp b/src/extension/internal/pdfinput/svg-builder.cpp
index d449d1a..4bbc7e8 100644
--- a/src/extension/internal/pdfinput/svg-builder.cpp
+++ b/src/extension/internal/pdfinput/svg-builder.cpp
@@ -1020,7 +1020,11 @@ void SvgBuilder::updateFont(GfxState *state) {
     GfxFont *font = state->getFont();
     // Store original name
     if (font->getName()) {
+#if POPPLER_CHECK_VERSION(0,72,0)
+        _font_specification = font->getName()->c_str();
+#else
         _font_specification = font->getName()->getCString();
+#endif
     } else {
         _font_specification = "Arial";
     }
@@ -1047,7 +1051,11 @@ void SvgBuilder::updateFont(GfxState *state) {
 
     // Font family
     if (font->getFamily()) { // if font family is explicitly given use it.
+#if POPPLER_CHECK_VERSION(0,72,0)
+        sp_repr_css_set_property(_font_style, "font-family", font->getFamily()->c_str());
+#else
         sp_repr_css_set_property(_font_style, "font-family", font->getFamily()->getCString());
+#endif
     } else { 
         int attr_value = 1;
         sp_repr_get_int(_preferences, "localFonts", &attr_value);
-- 
2.20.1

